### PR TITLE
Initial implementation of a Unit XBlock [LX-104] [SE-437]

### PR DESCRIPTION
**Pull request by bradenmacdonald**
Phew, the amount of boilerplate needed to get this XBlock with < 100 lines of code set up with most of the correct tooling is _ridiculous_!

# What it is

A minimal, pure-XBlock substitute for `vertical`. (`vertical` has a _lot_ of tech debt and mixins that can make development of new runtimes etc. messier than necessary, e.g. it has totally different XML parsing behavior than XBlocks do, because it uses XModule XML parsing code.)

# Test instructions:

You can run `make dev.run` then go to http://localhost:8000 to see it. Tests run in CircleCI and cover python 2 and 3.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Set up basic python environment:
+FROM phusion/baseimage:latest
```
Comment (symbolist) [inline Dockerfile:2]: Given that all these packages need to run as part of edxapp, we really should have a standard Docker container with everything set up.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+dev.run: dev.clean dev.build ## Clean, build and run test image
+	docker run -p 8000:8000 -v $(CURDIR):/usr/local/src/$(REPO_NAME) --name $(REPO_NAME)-dev $(REPO_NAME)-dev
+
+## Localization targets
+
+extract_translations: symlink_translations ## extract strings to be translated, outputting .po files
+	cd $(PACKAGE_NAME) && i18n_tool extract
+	mv $(EXTRACTED_DJANGO) $(EXTRACTED_TEXT)
+	#if [ -f "$(EXTRACTED_DJANGOJS)" ]; then cat $(EXTRACTED_DJANGOJS) >> $(EXTRACTED_TEXT); rm $(EXTRACTED_DJANGOJS); fi
```
Comment (symbolist) [inline Makefile:32]: There are a few commented lines in this file.

**Review by symbolist — APPROVED** [review]
Looks good to go! 

- [x] I tested this by running the dev container.
- [x] I read through the code
- [x] I checked for accessibility issues
- [x] Includes documentation

**Comment by symbolist** [comment]
@bradenmacdonald There are a few deprecation warnings https://circleci.com/gh/open-craft/unit-xblock/7 that I think can be fixed.

File: Dockerfile (hunk 1 above)
Comment (bradenmacdonald) [reply Dockerfile:2]: Yeah... This is to run `xblock-sdk`, not `edxapp`. And there is [a standard container for xblock-sdk](https://hub.docker.com/r/jbarciauskas/xblock-sdk/) kind of, but it's two years out of date and so wasn't usable with newer package versions.

File: Makefile (hunk 2 above)
Comment (bradenmacdonald) [reply Makefile:32]: It's because we're not using i18n right now but presumably will need to reinstate that in the future.

**Pull request merged**